cores/esp8266/PolledTimeout.h

```diff
@@ -23,9 +23,10 @@
  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-#include <limits>
-
-#include <Arduino.h>
+#include <c_types.h>               // IRAM_ATTR
+#include <limits>                  // std::numeric_limits
+#include <type_traits>             // std::is_unsigned
+#include <core_esp8266_features.h>
```

Comment:
Instead of features.h, this should be something like timing.h, with only the timing-specific functions, i.e.: split of fthe timing functions into separate timing.h|cpp files.

Reply:
After discussion, this is left for a different PR.